### ISSUE TITLE
coverlet.collector v3.1.0 causes fatal package downgrade of `Microsoft.Extensions.DependencyInjection`

### ISSUE THREAD

**Issue by TonyTroeff** [coverlet-coverage/coverlet#1270]
In some of my tests my team needs to start an instance of one of our services. This is a part of the code:
```C#
var hostBuilder = CreateHostBuilder(args);
var host = hostBuilder.Build();
await host.ExecuteStartupTasksAsync();

await host.RunAsync();
```

Unfortunatelly, after publishing the tests project the reference to Microsoft.EXtensions.DependencyInjection is downgraded to 2.2.0 and we receive the following exception upon tests execution:
```
Error Message:
   System.MissingMethodException : Method not found: 'Void Microsoft.Extensions.DependencyInjection.ServiceProviderOptions.set_ValidateOnBuild(Boolean)'.
  Stack Trace:
     at Microsoft.Extensions.Hosting.Host.<>c.<CreateDefaultBuilder>b__1_3(HostBuilderContext context, ServiceProviderOptions options)
   at Microsoft.Extensions.Hosting.HostingHostBuilderExtensions.<>c__DisplayClass3_0.<UseDefaultServiceProvider>b__0(HostBuilderContext context)
   at Microsoft.Extensions.Hosting.Internal.ServiceFactoryAdapter`1.CreateBuilder(IServiceCollection services)
   at Microsoft.Extensions.Hosting.HostBuilder.CreateServiceProvider()
   at Microsoft.Extensions.Hosting.HostBuilder.Build()
```

This property `ValidateOnBuild` is added after `Microsoft.Extensions.DependencyInjection` version `3.0.0` so it really could not be found.

I would be very interested to understand why `coverlet` should reference this package and how could we potentially fix our issue without removing `coverlet` as this is something we would not like to do.

**Comment by petli** [comment]
This is mostly a guess, but it may be similar to the issue with Newtonsoft.Json during publish that was fixed by #1243, so maybe that PR will also fix this case. It's not released yet, but should be in the nightly build if you want to give it a try:
https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/ConsumeNightlyBuild.md

**Comment by MarcoRossignoli** [comment]
@petli is right, can you try the nightly build and let us know if it's solved?
So we can also do a release asap.

**Comment by MarcoRossignoli** [comment]
Solved into shipped 3.1.1
